packages/babel-plugin-strip-runtime/src/__tests__/jsx-pragma.test.ts

```diff
@@ -0,0 +1,370 @@
+
+      for (const c of [code, codeWithRenamedImport]) {
+        expect(() =>
+          transform(c, {
+            run: 'both',
+            runtime: 'classic',
+            babelJSXPragma: 'jsx',
+          })
+        ).toThrow(/Found a `jsx` function call/);
+      }
+    });
+  });
+});
+
+describe('Compiled setup with automatic runtime', () => {
+  describe('with JSX pragma', () => {
+    describe('if Compiled is used in file', () => {
+      it('imports JSX runtime from React, not Compiled', () => {
+        const code = `
+          /** @jsxImportSource @compiled/react */
+          import { css } from '@compiled/react';
+
+          const Component = () => (
+            <div css={{ fontSize: 12, color: 'blue' }}>
+              hello world 2
+            </div>
+          );
+
+          const Component2 = () => (
+            <div css={css({ fontSize: 12, color: 'pink' })}>
+              hello world 2
+            </div>
+          );
+        `;
+
+        const actual = transform(code, {
+          run: 'both',
+          runtime: 'automatic',
+        });
+
+        expect(actual).toContain("import { jsx as _jsx } from 'react/jsx-runtime';");
+        // jsx function calls from React have "PURE" comments
+        // added beforehand from @babel/preset-react. This helps
+        // with tree-shaking by Webpack or Parcel
+        expect(actual).toContain('/*#__PURE__*/ _jsx');
+
+        expect(actual).not.toContain(
+          "import { jsxs as _jsxs } from '@compiled/react/jsx-runtime';"
+        );
+        expect(actual).not.toContain("import { jsx as _jsx } from '@compiled/react/jsx-runtime';");
+      });
+    });
+
+    describe('if Emotion is used in file', () => {
+      it('does not process file', () => {
+        const code = `
+          /** @jsxImportSource @emotion/react */
+          import { css } from '@emotion/react';
+
+          const Component = () => (
+            <div css={{ fontSize: 12, color: 'blue' }}>
+              hello world 2
+            </div>
+          );
+
+          const Component2 = () => (
+            <div css={css({ fontSize: 12, color: 'pink' })}>
+              hello world 2
+            </div>
+          );
+        `;
+
+        const actual = transform(code, {
+          run: 'both',
+          runtime: 'automatic',
+        });
+
+        expect(actual).toContain("import { jsx as _jsx } from '@emotion/react/jsx-runtime';");
+
+        expect(actual).not.toContain(
+          "import { jsxs as _jsxs } from '@compiled/react/jsx-runtime';"
+        );
+        expect(actual).not.toContain("import { jsx as _jsx } from '@compiled/react/jsx-runtime';");
+
+        // If any of the below statements are present, that means we accidentally
+        // got rid of the /** @jsxImportSource @emotion/react */ and the default React
+        // JSX runtime is being used instead of Emotion -- whoops!
+        expect(actual).not.toContain("import { jsxs as _jsxs } from 'react/jsx-runtime';");
```

Comment:
We could just check for the module as formatting / exports could change over time.